src/subscriptions/metamask.ts

```diff
@@ -0,0 +1,71 @@
+import { MetamaskProvider } from '../provider/types'
+
+export const MetamaskSubscriptionsSymbol = Symbol('MetamaskSubscriptions')
+
+export interface MetamaskSubscriptions {
+    [MetamaskSubscriptionsSymbol]: true
+    onAccountsChanged(callback: (accounts: string[]) => void, once?: boolean): () => void
+    onNetworkChanged(callback: (networkId: string) => void, once?: boolean): () => void
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    onData(callback: (error: Error | null, payload: any) => void, once?: boolean): () => void
+    onError(callback: (message: string) => void, once?: boolean): () => void
+}
+
+const createMetamaskSubscriptions = (provider: MetamaskProvider): MetamaskSubscriptions => {
+    const onAccountsChanged: MetamaskSubscriptions['onAccountsChanged'] = (cb, once) => {
+        if (once) {
+            provider.once('accountsChanged', cb)
+        } else {
+            provider.on('accountsChanged', cb)
+        }
+
+        return () => {
+            provider.off('accountsChanged', cb)
+        }
```

Comment:
nit: would it be too confusing to refactor all these repetitions into a helper function? Such as:

```typescript
function subscribeProviderToEvent(provider: MetamaskProvider, eventName:string, cb: ..., once?:boolean): ()=>void {
  if (once) {
    provider.once(eventName, cb)
  } else {
    provider.on(eventName, cb)
  }

  return () => {
    provider.off(eventName, cb)
  }
}
```

Reply:
It would be wrong to stuff all params into one function.
Can do this though:
```ts
const generateFunction = event = (cb, once) => {
       if (once) {
            provider.once(event, cb)
        } else {
            provider.on(event, cb)
        }

        return () => {
            provider.off('accountsChanged', cb)
        }

return {
  onAccountsChanged: MetamaskSubscriptions['onAccountsChanged'] = generateFunction('accountsChanged'),
  ...
```

would you prefer that?

Reply:
Yep, much cleaner :)

Reply:
Done